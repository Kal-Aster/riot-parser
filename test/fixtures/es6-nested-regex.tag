<es6-nested-regex>
  <h1>{ message }</h1>

  <script>
    const x = 1

    this.message = `
      regex: ${
        /x}`/.source
      }`

    this.foobar = `
    foobar() {
      return ${
        foo( 5 /x/.1 )  // not regex
      }
    }
`

  foo(bar) {}

  </script>
</es6-nested-regex>
