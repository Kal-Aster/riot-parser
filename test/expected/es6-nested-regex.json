{
  "html": {
    "type": 1,
    "name": "es6-nested-regex",
    "start": 0,
    "end": 290,
    "nodes": [
      {
        "type": 1,
        "name": "h1",
        "start": 21,
        "end": 41,
        "nodes": [
          {
            "type": 3,
            "text": "{ message }",
            "start": 25,
            "end": 36,
            "expr": [
              {
                "text": " message ",
                "start": 25,
                "end": 36
              }
            ],
            "parts": [
              "",
              "message"
            ]
          }
        ]
      }
    ]
  },
  "css": null,
  "js": {
    "type": 1,
    "name": "script",
    "start": 45,
    "end": 270,
    "text": {
      "type": 3,
      "text": "\n    const x = 1\n\n    this.message = `\n      regex: ${\n        /x}`/.source\n      }`\n\n    this.foobar = `\n    foobar() {\n      return ${\n        foo( 5 /x/.1 )  // not regex\n      }\n    }\n`\n\n  foo(bar) {}\n\n  ",
      "start": 53,
      "end": 261
    }
  }
}
